bundles/org.eclipse.osgi/supplement/src/org/eclipse/osgi/report/resolution/ResolutionReport.java

```diff
@@ -193,4 +193,24 @@ public interface Listener {
 	 * @return a resolution report message.
 	 */
 	String getResolutionReportMessage(Resource resource);
+
```

Comment:
Why do we need an inner interface for this.  `ResolutionReport` should be considered a `ProviderType` and we can add new methods to it.  If there are risks of multiple impls then the new methods can use default implementations that return -1 like the javadoc says.

Reply:
It should be but it is not marked as those and the default is sadly consumer type :-\

Reply:
I would mark it it as provider type then.  Regardless, we can add default impl to interface types that are consumer type.  But it is wrong to offer this as a consumer type anyway.

Reply:
okay I can do this still then we probably need to either silence or enhance the API check (would be a good opportunity I think) to understand `@Provider` / `@Consumer` type.

Reply:
done.